src/mappings/Competition/mapping.ts

```diff
@@ -0,0 +1,87 @@
+import 'allocator/arena';
+
+import { BigInt, ByteArray, crypto, ipfs, Address } from '@graphprotocol/graph-ts';
+// Import event types from the Reputation contract ABI
+import {
+  NewCompetitionProposal, NewSuggestion, NewVote, Redeem, SnapshotBlock,
+} from '../../types/Competition/Competition';
+
+import { ContributionRewardExt } from '../../types/ContributionRewardExt/ContributionRewardExt';
+import { CompetitionProposal, CompetitionSuggestion, CompetitionVote, Proposal } from '../../types/schema';
+import { concat, eventId } from '../../utils';
+
+export function handleNewCompetitionProposal(event: NewCompetitionProposal): void {
+  let contributionRewardExt = ContributionRewardExt.bind(event.params._contributionRewardExt);
+  let avatar = contributionRewardExt.avatar();
+  let contributionRewardScheme = crypto.keccak256(concat(avatar, event.params._contributionRewardExt)).toHex()
+  let competitionProposal = new CompetitionProposal(event.params._proposalId.toHex());
+  competitionProposal.proposal = event.params._proposalId.toHex();
+  competitionProposal.contract = event.address;
+  competitionProposal.dao = avatar.toHex();
+  competitionProposal.numberOfWinners = event.params._numberOfWinners;
+  competitionProposal.rewardSplit = event.params._rewardSplit;
+  competitionProposal.startTime = event.params._startTime;
+  competitionProposal.votingStartTime = event.params._votingStartTime;
+  competitionProposal.suggestionsEndTime = event.params._suggestionsEndTime;
+  competitionProposal.endTime = event.params._endTime;
+  competitionProposal.numberOfVotesPerVoters = event.params._maxNumberOfVotesPerVoter;
+  competitionProposal.contributionReward = contributionRewardScheme;
+  competitionProposal.createdAt = event.block.timestamp;
+  competitionProposal.save();
+  let proposal = Proposal.load(competitionProposal.id);
+  if (proposal != null) {
+    proposal.competition = competitionProposal.id;
+    proposal.save();
+  }
+}
+
+export function handleRedeem(event: Redeem): void {
+  // TODO: Implement redeem and reward logic.
+}
+
+export function handleNewSuggestion(event: NewSuggestion): void {
+  let competitionSuggestion = new CompetitionSuggestion(
+    crypto.keccak256(
+      concat(
+        event.params._proposalId,
+        event.params._suggestionId as ByteArray,
+      ),
+    ).toHex(),
+  );
+  competitionSuggestion.proposal = event.params._proposalId.toHex();
+  competitionSuggestion.descriptionHash = event.params._descriptionHash;
+  let ipfsData = ipfs.cat('/ipfs/' + event.params._descriptionHash.toHex());
+  if (ipfsData != null) {
+    competitionSuggestion.description = ipfsData.toString();
```

Comment:
this should have title as well( same as normal proposal IPFS data).
I guess we should use the same code as in proposal Entity for that. even just call it to parse the ipfs data to `title` and `description`

Reply:
@jellegerbrandy specification does not include a title property. Are you sure it's part of this here?

Reply:
yes

Reply:
I'm sorry I missed that, i changed the issue. A "suggestion"  has a title, a description, and a URL, all optional, and all stored on IPFS, under "descriptionHash"